### ISSUE TITLE
[BUG] Nexo Service: crypto error (3) when requesting a payment on POS with local integration

### ISSUE THREAD

**Issue by simonszalai** [Adyen/adyen-node-api-library#762]
**Describe the bug**
When trying to send a local payment request the same way as the unit tests do ([here](https://github.com/Adyen/adyen-node-api-library/blob/4f3a84c63ed52ef84475a149bd6859f50dd7850b/src/__mocks__/base.ts) and [here](https://github.com/Adyen/adyen-node-api-library/blob/4f3a84c63ed52ef84475a149bd6859f50dd7850b/src/__tests__/terminalLocalAPI.spec.ts#L48)), I get the following error: 
```
Error: {"errors": ["Nexo Service: crypto error (3)"], "ServiceID": "-NULL-"}
    at IncomingMessage.<anonymous> (/Users/user1/adyen_api/node_modules/@adyen/api-library/src/httpClient/httpURLConnectionClient.ts:163:45)
    at IncomingMessage.emit (events.js:412:35)
    at IncomingMessage.emit (domain.js:470:12)
    at endReadableNT (internal/streams/readable.js:1317:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
```
I configured a proper security key in Adyen's Customer Area and used that in the request.

**To Reproduce**
Execute the code below with `ts-node` after replacing the following placeholders with appropriate values:
[TERMINAL_ID]: ID of the terminal where the request should be sent, (e.g.: "P400Plus-503026317")
[SECURITY_KEY_ID]: ID of the security key configured in Adyen's customer area
[SECURITY_KEY_PASS: passphrase of the security key configured in Adyen's customer area
[ADYEN_API_KEY]: Test API key for Adyen
[ADYEN_MERCH_ACC]: Adyen merchant account name
[TERMINAL_LOCAL_ENDPOINT]: Local IP address of the terminal (e.g.: "https://192.168.86.230")

Also download [Adyen's root test certificate](https://docs.adyen.com/point-of-sale/choose-your-architecture/local#install-root-cert) and save it as `./adyen-terminalfleet-test.pem`

```
import { Client, Config, TerminalLocalAPI, Modification } from "@adyen/api-library"
import {
    TerminalApiRequest,
    MessageHeader,
    SaleToPOIRequest,
    PaymentTransaction,
    PaymentRequest,
    AmountsReq,
    SaleData,
    TransactionIdentification,
    MessageCategoryType,
    MessageClassType,
    MessageType,
    SecurityKey,
} from "@adyen/api-library/lib/src/typings/terminal/models"

const id = Math.floor(Math.random() * Math.floor(10000000)).toString()
const getMessageHeader = ({ messageCategory = MessageCategoryType.Payment }: { messageCategory?: MessageCategoryType } = {}): MessageHeader => ({
    MessageCategory: messageCategory,
    MessageClass: MessageClassType.Service,
    MessageType: MessageType.Request,
    POIID: [TERMINAL_ID],
    ProtocolVersion: "3.0",
    SaleID: id,
    ServiceID: id,
})

const timestamp = (): string => new Date().toISOString()
const transactionIdentification: TransactionIdentification = {
    TimeStamp: timestamp(),
    TransactionID: id,
}

const saleData: SaleData = {
    SaleTransactionID: transactionIdentification,
    SaleToAcquirerData: {
        applicationInfo: {
            merchantApplication: {
                version: "1",
                name: "test",
            },
        },
        metadata: {
            someMetaDataKey1: "YOUR_VALUE",
            someMetaDataKey2: "YOUR_VALUE",
        },
    },
}

const amountsReq: AmountsReq = {
    Currency: "CAD",
    RequestedAmount: 1,
}
const paymentTransaction: PaymentTransaction = {
    AmountsReq: amountsReq,
}

const paymentRequest: PaymentRequest = {
    PaymentTransaction: paymentTransaction,
    SaleData: saleData,
}

const getSaleToPOIRequest = (messageHeader: MessageHeader, request: Partial<SaleToPOIRequest>): SaleToPOIRequest => ({
    MessageHeader: messageHeader,
    ...request,
})

const createTerminalAPIPaymentRequest = (): TerminalApiRequest => {
    const messageHeader = getMessageHeader()
    const saleToPOIRequest = getSaleToPOIRequest(messageHeader, { PaymentRequest: paymentRequest })
    return { SaleToPOIRequest: saleToPOIRequest }
}

const terminalAPIPaymentRequest = createTerminalAPIPaymentRequest()

const securityKey: SecurityKey = {
    AdyenCryptoVersion: 1,
    KeyIdentifier: [SECURITY_KEY_ID],
    KeyVersion: 1,
    Passphrase: [SECURITY_KEY_PASS],
}

export const createClient = (apiKey = process.env.ADYEN_API_KEY): Client => {
    const config: Config = new Config()
    config.apiKey = [ADYEN_API_KEY]
    config.merchantAccount = [ADYEN_MERCH_ACC]

    const terminalId = [TERMINAL_ID]
    config.terminalApiLocalEndpoint = [TERMINAL_LOCAL_ENDPOINT]

    config.certificatePath = "./adyen-terminalfleet-test.pem"

    const client = new Client({ config })
    client.setEnvironment("TEST")

    return client
}

const client = createClient()
const terminalLocalAPI = new TerminalLocalAPI(client)

const run = async () => {
    try {
        await terminalLocalAPI.request(terminalAPIPaymentRequest, securityKey)
    } catch (e) {
        console.log(e)
    }
}

run()
```

**Expected behavior**
The terminal is able to decrypt the message and initiate the payment.

**Desktop (please complete the following information):**
 - OS: MacOS 11.5.2
 - Node Version: v14.17.4
 - NPM Version: 6.14.14

**Comment by wboereboom** [comment]
Hi @simonszalai,

Thank you for raising this issue, and apologies for the delayed response.

I was able to reproduce your issue on a terminal, and I can confirm the problem indeed lies in our library.
The fix is on its way and will be released early next week.

Kind Regards
Wouter
Adyen

**Comment by simonszalai** [comment]
Thanks a lot, looking forward to the fix!

**Comment by wboereboom** [comment]
Hi @simonszalai ,

This issue is fixed in the latest release of the library (9.1.0).

Regards,
Wouter
Adyen
